L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp1.cc

```diff
@@ -171,8 +173,42 @@ void HGCalTriggerGeometryV9Imp1::initialize(const HGCalGeometry* hgc_ee_geometry
   }
   last_trigger_layer_ = trigger_layer - 1;
   fillMaps();
-  fillNeighborMap(l1tCellNeighborsMapping_, trigger_cell_neighbors_, false);        // silicon
-  fillNeighborMap(l1tCellNeighborsSciMapping_, trigger_cell_neighbors_sci_, true);  // scintillator
+  fillNeighborMap(l1tCellNeighborsMapping_, trigger_cell_neighbors_,
+                  false);  // silicon
+  fillNeighborMap(l1tCellNeighborsSciMapping_, trigger_cell_neighbors_sci_,
+                  true);  // scintillator
+  fillInvalidTriggerCells();
+}
+
+void HGCalTriggerGeometryV9Imp1::initialize(const HGCalGeometry* hgc_ee_geometry,
+                                            const HGCalGeometry* hgc_hsi_geometry,
+                                            const HGCalGeometry* hgc_hsc_geometry,
+                                            const HGCalGeometry* hgc_nose_geometry) {
+  throw cms::Exception("BadGeometry") << "HGCalTriggerGeometryV9Imp1 geometry cannot be initialized with the "
+                                         "HFNose geometry; use HGCalTriggerGeometryV9Imp2";
+
+  setEEGeometry(hgc_ee_geometry);
+  setHSiGeometry(hgc_hsi_geometry);
+  setHScGeometry(hgc_hsc_geometry);
+  heOffset_ = eeTopology().dddConstants().layers(true);
+  totalLayers_ = heOffset_ + hsiTopology().dddConstants().layers(true);
+  trigger_layers_.resize(totalLayers_ + 1);
+  trigger_layers_[0] = 0;  // layer number 0 doesn't exist
+  unsigned trigger_layer = 1;
+  for (unsigned layer = 1; layer < trigger_layers_.size(); layer++) {
+    if (disconnected_layers_.find(layer) == disconnected_layers_.end()) {
+      // Increase trigger layer number if the layer is not disconnected
+      trigger_layers_[layer] = trigger_layer;
+      trigger_layer++;
+    } else {
+      trigger_layers_[layer] = 0;
+    }
+  }
+  fillMaps();
+  fillNeighborMap(l1tCellNeighborsMapping_, trigger_cell_neighbors_,
+                  false);  // silicon
+  fillNeighborMap(l1tCellNeighborsSciMapping_, trigger_cell_neighbors_sci_,
+                  true);  // scintillator
```

Comment:
Now that there is an exception thrown just above, there is no need to have this code

Reply:
remove